modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java

```diff
@@ -715,9 +716,17 @@ private Map<String, Object> _toProperties(
 				if ((dtoConverterContext.getLocale() != null) &&
 					(objectField_i18n != null)) {
 
-					serializable = GetterUtil.getString(
+					String objectFieldValue = GetterUtil.getString(
 						objectField_i18n.get(
 							String.valueOf(dtoConverterContext.getLocale())));
+
+					if (objectFieldValue.equals(StringPool.BLANK)) {
+						objectFieldValue = GetterUtil.getString(
+							objectField_i18n.get(
+								String.valueOf(LocaleUtil.getSiteDefault())));
+					}
+
+					serializable = objectFieldValue;
```

Comment:
This should be validated by BPM guys as there might be cases we are not seeing, affecting expectations in other parts using the DTO. 

Could we handle this from the fragment? If we grab the values of the sort entity we can examine the locale availability and send the default we want. This way we could use a consistent process in the DSM, not affecting the rest.

If this turns out to be an object bug, we can still keep our logic in the fragment or remove it, depending of our requirements. Admittedly, ideally we'd like to use object DTO conversion logic as it is.

Makes sense?

Reply:
I can update [FDSViewFragmentRenderer.java#L985](https://github.com/liferay/liferay-portal/blob/0b71fc507ba9a54b7cd01562e31133d2d26b06fa/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java#L985) to grab the appropriate translated label which will fix the blank values in the FDS order dropdown.

But the DSM sort page relies directly on the response of the [fetch in Sorting.tsx](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/data_set/sorting/Sorting.tsx#L452-L454) so that will need to be on the objects end. I can however in the meantime provide a custom cell renderer to fallback to the site default in `label_i18n` that would go here [Sorting.tsx#L638-L642](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/data_set/sorting/Sorting.tsx#L638-L642)